case-lib/hijack.sh

```diff
@@ -77,6 +78,7 @@ function exit()
         ;;
     esac
 
+    trap '' EXIT
```

Comment:
I don't think you need this.

Reply:
I missed all the processing done on the exit code, sorry. I don't think you don't have access to the exit code in `trap EXIT`, so I don't think this can work. How did you test it?

Reply:
yes, you are right, I already test it without this line, it can be worked without recursive call the trap function

Reply:
@Bin-QA I don't understand your last comment, I don't understand how this can work and I would like to know with what failure and  non-zero exit you tested this change because I really don't understand how this can report failures. Try this:

```shell
trap 'func_exit_handler' EXIT

func_exit_handler()
{
    printf '$1 = |%s|\n' "$1"
    local exit_status=${1:-0}
    printf '(wrong) exit status  = |%s|\n' "${exit_status}"
}
exit 5
```
This prints:
```
$1 = ||
(wrong) exit status  = |0|
```
Cc: @xiulipan

Reply:
Sorry for miss your point

Reply:
@marc-hb let's track it at gitlab